src/lib.rs

```diff
@@ -31,6 +31,69 @@ pub fn kernel_main(bootinfo: &impl BootInfo) -> ! {
         tracing::dispatcher::set_global_default(subscriber).unwrap();
     }
 
+    if let Some(mut framebuf) = bootinfo.framebuffer() {
+        use hal_core::framebuffer::{Draw, RgbColor};
+        tracing::trace!("framebuffer exists!");
+        framebuf.clear();
+        let mut color = 0;
+        for row in 0..framebuf.height() {
+            color += 1;
+            match color {
+                1 => {
+                    framebuf.fill_row(row, RgbColor::RED);
+                }
+                2 => {
+                    framebuf.fill_row(row, RgbColor::GREEN);
+                }
+                3 => {
+                    framebuf.fill_row(row, RgbColor::BLUE);
+                    color = 0;
+                }
+                _ => {}
+            }
+        }
+        tracing::trace!("made it grey!");
+
+        use embedded_graphics::{
+            mono_font::{ascii, MonoTextStyle},
+            pixelcolor::{Rgb888, RgbColor as _},
+            prelude::*,
+            text::{Alignment, Text},
+        };
+        let center = Point::new(
+            (framebuf.width() / 2) as i32,
+            (framebuf.height() / 2) as i32,
+        );
+        let mut target = framebuf.clear().as_draw_target();
+        let small_style = MonoTextStyle::new(&ascii::FONT_6X10, Rgb888::WHITE);
+        Text::with_alignment(
+            "the glorious\n",
+            Point::new(center.x, center.y - 20),
+            small_style,
+            Alignment::Center,
+        )
+        .draw(&mut target)
+        .expect("never panics");
+        Text::with_alignment(
+            "MYCELIUM\n",
+            center,
+            MonoTextStyle::new(&ascii::FONT_10X20, Rgb888::WHITE),
+            Alignment::Center,
+        )
+        .draw(&mut target)
+        .expect("never panics");
+        Text::with_alignment(
+            concat!("operating system\n\n v", env!("CARGO_PKG_VERSION"), "\n"),
+            Point::new(center.x, center.y + 10),
+            small_style,
+            Alignment::Center,
+        )
+        .draw(&mut target)
+        .expect("never panics");
+
+        tracing::trace!("hahahaha yayyyy we drew a screen!");
+    }
+
     arch::interrupt::init::<arch::InterruptHandlers>();
```

Comment:
I don't love that were setting up interrupts after doing test printing on the framebuffer. I feel like we should initialize it after interrupts, so that things like faults are caught properly.

Reply:
hmm, but without the framebuffer we can't log anything or display an oops screen on fautls? so i think we should just "try really hard not to fault here"